spec/ddtrace_integration_spec.rb

```diff
@@ -14,6 +14,13 @@ def wait_for_tracer_sent
       try_wait_until { Datadog.tracer.writer.transport.stats.success > 0 }
     end
 
+    before do
+      # Ensure tracer is not running before test starts.
+      # This shouldn't be necessary if all other tests
+      # always terminate the tracer.
+      Datadog.shutdown!
+    end
```

Comment:
I don't quite like making the tests being defensive in this way, e.g. if the tracer shouldn't stay around between tests, I'd favor something like the thread leak detector where we blame the test that forgot to clean up, rather than make other tests need to clean up after it :)

Reply:
We have a `Datadog.shutdown!` helper that runs after every test: https://github.com/DataDog/dd-trace-rb/blob/master/spec/spec_helper.rb#L157-L169

This `before` block introduced here can only catch any issues if the tracer was initialized outside of the scope of an RSpec test run, which seems to be happening.

I added it because I noticed that tests this PR was flaky without it, and I leaned towards non-flakiness at the expense of masking the root cause. I'll revert this change then, and see if we can spot the root cause at a later time.

Reply:
The missing shutdown was in the same file: [`50c9a0b` (#1334)](https://github.com/DataDog/dd-trace-rb/pull/1334/commits/50c9a0bf6cbae482f44edb25ec0f8961cb79360f). Fixed it.